lib/veri.ml

```diff
@@ -289,15 +248,28 @@ class ['a] t arch dis =
         | Error er -> SM.update (fun c -> c#notify_error er)
         | Ok insn -> self#eval_insn insn
 
+    method! eval_memory_load mv =
+      SM.get () >>= fun init ->
+      let is_new e = not (Set.mem init#events e) in
+      self#eval_memory_store mv  >>= fun _ ->
+      SM.update (fun c -> c#discard_event is_new) >>= fun () ->
```

Comment:
We shouldn't emit this event on the first hand. Use `super#eval_store` to perform the side-effect of storing.